polydodo/src/components/d3component.js

```diff
@@ -1,23 +1,9 @@
-import React from "react";
-import PropTypes from "prop-types";
+import React, { useEffect, useRef } from "react";
 
-class D3Component extends React.Component {
-  static propTypes = {
-    callback: PropTypes.func.isRequired,
-  }
+const D3Component = React.memo(({ callback }) => {
```

Comment:
Use React.memo so we don't rerender when the parent does rerender with that same callback prop. We want that since D3 will handle DOM updates inside this svg.